hypergan/graph/graph.py

```diff
@@ -33,14 +33,24 @@ def discriminator(self, x, f,z,g,gz):
         single_batch_size = config.batch_size
         channels = config.channels
         # combine to one batch, per Ian's "Improved GAN"
+        print('x,g', x,g[0])
+        x = tf.reshape(x, [config.batch_size, -1, 1, 1])
+        g[-1] = tf.reshape(g[-1], [config.batch_size, -1, 1, 1])
+
         xs = [x]
         gs = g
         graph.xs=xs
         graph.gs=gs
         g = g[-1]
         if len(gs) > 1:
             for i in gs:
-                resized = tf.image.resize_images(xs[-1],[int(xs[-1].get_shape()[1]//2),int(xs[-1].get_shape()[2]//2)], 1)
+                s = [int(elem) for elem in xs[-1].get_shape()]
```

Comment:
Lets just refactor this whole thing.  If we find calling resize_images on x is costly we can add a cache.